bundles/org.openhab.binding.qbus/src/main/java/org/openhab/binding/qbus/internal/protocol/QbusCommunication.java

```diff
@@ -0,0 +1,913 @@
+    private @Nullable PrintWriter qOut;
+    private @Nullable BufferedReader qIn;
+
+    private boolean listenerStopped;
+    private boolean qEventsRunning;
+
+    private Gson gsonOut = new Gson();
+    private Gson gsonIn;
+
+    private @Nullable String CTD;
+
+    private boolean CTDConnected = false;
+
+    private final Map<Integer, QbusBistabiel> bistabiel = new HashMap<>();
+    private final Map<Integer, QbusScene> scene = new HashMap<>();
+    private final Map<Integer, QbusDimmer> dimmer = new HashMap<>();
+    private final Map<Integer, QbusRol> rol = new HashMap<>();
+    private final Map<Integer, QbusThermostat> thermostat = new HashMap<>();
+    private final Map<Integer, QbusCO2> co2 = new HashMap<>();
+
+    private @Nullable QbusBridgeHandler bridgeCallBack;
+
+    /**
+     * Constructor for Qbus communication object, manages communication with
+     * Qbus Server.
+     *
+     */
+    public QbusCommunication() {
+        GsonBuilder gsonBuilder = new GsonBuilder();
+        gsonBuilder.registerTypeAdapter(QbusMessageBase.class, new QbusMessageDeserializer());
+        gsonIn = gsonBuilder.create();
+    }
+
+    /**
+     * Start communication with Qbus Server, run through initialization and start thread listening
+     * to all messages coming from Qbus.
+     *
+     * @param addr : IP-address of Qbus Server
+     * @param port : Communication port of Qbus server
+     * @throws InterruptedException
+     * @throws IOException
+     *
+     */
+
+    public synchronized void startCommunication() throws IOException, InterruptedException {
+        QbusBridgeHandler handler = bridgeCallBack;
+        CTDConnected = false;
+
+        if (qEventsRunning) {
+            throw new IOException("Previous listening thread is still active.");
+        }
+
+        if (handler == null) {
+            throw new IOException("No Bridge handler initialised.");
+        }
+
+        InetAddress addr = InetAddress.getByName(handler.getAddress());
+        Integer port = handler.getPort();
+
+        if (port != null) {
+            Socket socket = new Socket(addr, port);
+            qSocket = socket;
+            qOut = new PrintWriter(socket.getOutputStream(), true);
+            qIn = new BufferedReader(new InputStreamReader(socket.getInputStream()));
+            logger.debug("Connected via local port {} from thread {}", socket.getLocalPort(),
+                    Thread.currentThread().getId());
+        } else {
+            return;
+        }
+
+        setSN();
+        getSN();
+
+        // Connect to Qbus server
+        Connect();
+
+        // If Qbus Client is connected then initialize and start listening for incoming commands, else put Bridge
+        // offline
+        if (CTDConnected == true) {
+            initialize();
+            Thread thread = new Thread(this::qEvents);
+            thread.setName("OH-binding");
+            thread.setDaemon(true);
+            thread.start();
```

Comment:
The Thread's name should include the Thing UID. You could use `NamedThreadFactory` to retrieve a correctly configured Thread. Here are some examples: https://github.com/openhab/openhab-addons/search?q=NamedThreadFactory